doc/devguide/extending/app-layer/app-layer-frames.rst

```diff
@@ -0,0 +1,225 @@
+*******************************
+Application Layer Frame Support
+*******************************
+
+.. contents:: Table of Contents
+
+Baseline
+========
+
+- `Suricata rules format <https://suricata.readthedocs.io/en/latest/rules/intro.html>`_
+- It helps to understand the basics of the `Multi-pattern rule matching <https://suricata.readthedocs.io/en/latest/configuration/suricata-yaml.html?highlight=multi%20pattern%20matcher#detection-engine>`_ algorithm, used for rule filtering and to address engine performance
```

Comment:
Not really sure ppl really need to understand MPM. Doesn't seem they need to read this first to understand frames.